lib/src/injector_base.dart

```diff
@@ -6,49 +6,45 @@ import 'package:injector/src/factory/provider_factory.dart';
 import 'package:injector/src/factory/singelton_factory.dart';
 
 class Injector {
-  /**
-   * The static / single instance of the injector
-   */
+  /// The static/single instance of the [Injector].
   static final Injector appInstance = Injector();
 
-  /**
-   * The constructor to create a new instance
-   */
-  Injector();
-
-  /**
-   * The Map that contains:
-   *
-   *  [SingletonFactory] - The single instances
-   *
-   * [ProviderFactory] - The provider factory that contains the type information.
-   *                     If creates a new instance every time
-   */
+  /// Stores [SingletonFactory] and [ProviderFactory] instances that have been
+  /// registered by [registerSingleton] and [registerDependency] respectively.
   Map<String, Factory<Object>> _factoryMap = Map<String, Factory<Object>>();
 
-  /**
-   * abstract class UserService {
-   *  void Login(String username, String password);
-   * }
-   *
-   * class UserServiceImpl implements UserService {
-   *  void Login(String username, String password){
-   *    .....
-   *    .....
-   *  }
-   * }
-   *
-   * Injector.appInstance.registerDependency<UserService>((_) => new UserService);
-   *
-   * If you now trying to get your dependency with:
-   *
-   *     var userService = Injector.appInstance.getDependency<UserService>();
-   *
-   * You get a new instance of your dependency every time that you call this function!
-   *
-   * [override] => Override the existing dependency
-   *
-   */
+  /// Registers a dependency by using [ProviderFactory].
+  ///
+  /// Overrides dependencies with the same signature when [override] is true.
+  /// Uses [dependencyName] to differentiate between dependencies that have the
+  /// same type.
+  ///
+  /// The signature of a dependency consists of [T]
+  /// and the optional [dependencyName].
+  ///
+  /// The injector will always create new instances of dependencies that got
+  /// registered this way when calling [getDependency].
+  ///
+  /// Example:
+  /// ```dart
+  /// abstract class UserService {
+  ///     void Login(String username, String password);
+  /// }
+  ///
+  /// class UserServiceImpl implements UserService {
+  ///     void Login(String username, String password) {
+  ///         // does something..
+  ///     }
+  /// }
+  ///
+  /// Injector.appInstance.registerDependency<UserService>((_) => new UserService);
```

Comment:
you need the brackets :o `()`
`new UserServiceImpl()`

You can not create an instance of an abstract class